### DISCUSSION TITLE
Unable to run even the most basic Change API subscription

### DISCUSSION THREAD

**Discussion by ItsRobbAllen** [ravendb/ravendb#16068] [Q&A] [answered]
I have never used the Change API, so to test in Linqpad, I just copy/pasted the code in the documentation

```
	IDisposable subscription = store
	.Changes()
	.ForDocument("employees/1")
	.Subscribe(
		change =>
		{
			switch (change.Type)
			{
				case DocumentChangeTypes.Put:
					// do something
					break;
				case DocumentChangeTypes.Delete:
					// do something
					break;
			}
		});
```

and it complains in the Subscribe method
`Cannot convert lambda expression to type 'IObserver<DocumentChange>' because it is not a delegate type`

Am I missing a reference somewhere as this seems like it is the most basic of examples?

*edit*

I can make this work by creating my own class that implements `IObserver<DocumentChange>` and passing that as the parameter, but it's just odd to me that the sample code doesn't work & makes me think I'm missing something simple.

**Comment by georgiosd** [comment, marked as answer]
Just add the `System.Reactive` nuget - `IObserver` comes from there!

**Reply by ItsRobbAllen** [reply]
Ah, that's at the very bottom of the documentation as a _suggestion_. It could be a little clearer that the examples require it, but thanks for your help, that makes everything work as the examples show.
